src/chainstate/coordinator/mod.rs

```diff
@@ -769,15 +769,27 @@ impl<
                     }
                     if !attachments_instances.is_empty() {
                         info!(
-                            "Atlas: {} attachment instances emitted from events",
-                            attachments_instances.len()
+                            "Atlas: New attachment instances emitted by block";
+                            "attachments_count" => attachments_instances.len(),
+                            "index_block_hash" => %block_receipt.header.index_block_hash(),
+                            "stacks_height" => block_receipt.header.stacks_block_height,
                         );
-                        match self.attachments_tx.send(attachments_instances) {
-                            Ok(_) => {}
-                            Err(e) => {
-                                error!("Atlas: error dispatching attachments {}", e);
+                        if let Some(atlas_db) = self.atlas_db.as_mut() {
```

Comment:
Correct me if I'm wrong, but doesn't this call to `queue_attachment_instance()` happen _outside_ of block-processing?  Like, if a node processed a block and then crashed before reaching this code path, would it be able to enqueue the block's attachment data on restart?

Reply:
Would it be possible to commit to the attachment data from the event stream from within `StacksChainState::append_block()`, right before we commit to the Stacks block?  Enqueuing attachment instances is idempotent, so re-doing them in the event the node crashes in-between saving them and committing the block should be safe.

Reply:
Yes, but I'd like to do that in a separate PR. It's true that this change doesn't make the atlas queue entirely crash proof, but it removes the 2-thread dependency for the atlas queue, which makes the shutdown behavior much better already.

This PR is already changing a lot of how Atlas works, so I'd prefer to not also move atlas queueing logic into `append_block` as part of the same changeset.